### PR TITLE
MULE-10744: HTTP Extension: Migrate TLS Tests

### PR DESCRIPTION
**Pull request by anosenzo**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -143,6 +143,7 @@
         <asmVersion>3.3.1</asmVersion>
         <aspectjVersion>1.8.5</aspectjVersion>
         <asyncHttpClientVersion>1.9.39</asyncHttpClientVersion>
+        <apacheHttpClientVersion>4.4.1</apacheHttpClientVersion>
```
Comment (aiannucci) [inline pom.xml]: Can we use the latest version?

File: tests/http/src/test/java/org/mule/test/module/http/functional/tls/HttpTlsContextCustomProtocolsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  @ClassRule
+  public static final DynamicPort httpsInternalTls11Port = new DynamicPort("https.internal.tlsv11");
+  @ClassRule
+  public static final DynamicPort httpsInternalTls12Port = new DynamicPort("https.internal.tlsv12");
+  @ClassRule
+  public static final DynamicPort httpsInternalAllTlsPort = new DynamicPort("https.internal.alltls");
+  @ClassRule
+  public static SystemProperty enabledProtocols = new SystemProperty("enabledProtocols", "TLSv1.2");
+  @ClassRule
+  public static SystemProperty verboseExpcetions = new SystemProperty("mule.verbose.exceptions", "true");
```
Comment (aiannucci) [inline tests/http/src/test/java/org/mule/test/module/http/functional/tls/HttpTlsContextCustomProtocolsTestCase.java]: fix typo

**Comment by anosenzo** [comment]
> Add all the other HTTP matchers to the HTTP matchers folder
Please merge MULE-10777 PR first with the matchers.

**Comment by aiannucci** [comment]
jenkins validate

**Comment by aiannucci** [comment]
jenkins validate

**Comment by anosenzo** [comment]
jenkins validate

**Pull request merged**